src/NVMLIOGroup.cpp

```diff
@@ -0,0 +1,673 @@
+
+#include <cmath>
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <sched.h>
+
+#include "IOGroup.hpp"
+#include "PlatformTopo.hpp"
+#include "NVMLDevicePool.hpp"
+#include "Exception.hpp"
+#include "Agg.hpp"
+#include "Helper.hpp"
+
+namespace geopm
+{
+    NVMLIOGroup::NVMLIOGroup()
+        : NVMLIOGroup(platform_topo(), nvml_device_pool(platform_topo().num_domain(GEOPM_DOMAIN_CPU)))
+    {
+    }
+
+    // Set up mapping between signal and control names and corresponding indices
+    NVMLIOGroup::NVMLIOGroup(const PlatformTopo &platform_topo, const NVMLDevicePool &device_pool)
+        : m_platform_topo(platform_topo)
+        , m_nvml_device_pool(device_pool)
+        , m_is_batch_read(false)
+        , m_signal_available({{"NVML::FREQUENCY", {
+                                  "Streaming multiprocessor frequency in hertz",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::average,
+                                  string_format_double
+                                  }},
+                              {"NVML::UTILIZATION_ACCELERATOR", {
+                                  "Percentage of time the accelerator operated on a kernel in the last set of driver samples",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::average,
+                                  string_format_double
+                                  }},
+                              {"NVML::POWER", {
+                                  "Accelerator power usage in watts",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::sum,
+                                  string_format_double
+                                  }},
+                              {"NVML::POWER_LIMIT", {
+                                  "Accelerator power limit in watts",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::sum,
+                                  string_format_double
+                                  }},
+                              {"NVML::FREQUENCY_MEMORY", {
+                                  "Accelerator memory frequency in hertz",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::average,
+                                  string_format_double
+                                  }},
+                              {"NVML::THROTTLE_REASONS", {
+                                  "Accelerator clock throttling reasons",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::average,
```

Comment:
Is reasons an enum?  If so it shouldn't be averaged.

Reply:
It is.  Changed to expect_same.